app/server_inactivity.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/mattermost/mattermost-server/v6/store"
+)
+
+const serverInactivityHours = 100
+
+func (s *Server) doInactivityCheck() {
+	if !s.Config().FeatureFlags.EnableInactivityCheckJob {
+		mlog.Info("No activity check because EnableInactivityCheckJob feature flag is disabled")
+		return
+	}
+
+	inactivityDurationHourseEnv := os.Getenv("MM_INACTIVITY_DURATION")
+	inactivityDurationHours, parseError := strconv.ParseFloat(inactivityDurationHourseEnv, 64)
+	if parseError != nil {
+		// default to 100 hours
+		inactivityDurationHours = serverInactivityHours
+	}
+
+	systemValue, sysValErr := s.Store.System().GetByName("INACTIVITY")
+	if sysValErr != nil {
+		// any other error apart from ErrNotFound we stop execution
+		if _, ok := sysValErr.(*store.ErrNotFound); !ok {
+			mlog.Warn("An error occurred while getting INACTIVITY from system store", mlog.Err(sysValErr))
+			return
+		}
+	}
+
+	// If we have a system value, it means this job already ran atleast once.
+	// we then check the last time the job ran plus the last time a post was made to determine if we
+	// can remind the user to use workspace again. If no post was made, we check the last time they logged in (session)
+	// and determine whether to send them a reminder.
+	if systemValue != nil {
+		sysT, _ := strconv.ParseInt(systemValue.Value, 10, 64)
+		tt := time.Unix(sysT/1000, 0)
+		timeLastSentInativityEmail := time.Since(tt).Hours()
+
+		lastPostAt, _ := s.Store.Post().GetLastPostRowCreateAt()
+		if lastPostAt != 0 {
+			posT := time.Unix(lastPostAt/1000, 0)
+			timeForLastPost := time.Since(posT).Hours()
+
+			if timeLastSentInativityEmail > inactivityDurationHours && timeForLastPost > inactivityDurationHours {
+				s.takeInactivityAction()
+			}
```

Comment:
@AGMETEOR can you confirm is the inactivity emails is sent again after the first pass of 100 hours of inactivity? cc/ @dontoisme